### DISCUSSION TITLE
[Kavita] Sharing media

### DISCUSSION THREAD

**Discussion by therobbiedavis** [Kareadita/Kavita#2575] [Ideas]
Essentially if someone wants to share just one issue/chapter or book, they can generate a secure link and send it to a friend without needing the friend to setup an account. Undecided if link would just be to the source file or if it would open a reader to consume from Kavita without tracking.

Admin:
- Can control if a user can and can not generate  a share link

User:
- Can see their shares, and deactivate/delete them (similar to discord invites)

URL: http[s]://[ip/reverseproxy]/share/[generated code]

**Comment by danwilldev** [comment]
Would it also be beneficial to have an expiry time setting for the link? So the user sharing can specify how long it can remain accessible

**Reply by DieselTech** [reply]
I think it would be. That way you don't have shares piling up over time and end up with potentially 100's of open shares going. 

As for admin controls there needs to be a "tree" like view of what users have what shares opened. That way you can see if your users are opening too many outsides shares. Yes I know the admin could just not let the normal user open links, but this is essentially the equivalent of letting normal users create "accounts" on the server. There needs to be a way to monitor / curb potential abuse. 

For instance ABS has the "RSS Feeds" page within the server settings so you can see all the open slugs

![image](https://github.com/Kareadita/Kavita/assets/30128380/9109e9ee-6597-4076-8fd1-f234c408e8c9)

From there you can also just delete an open link as well.
